### PR TITLE
Create 2023-state-of-rust-analysis.md

### PR DESCRIPTION
**Pull request by graciegregory**
Here is the analysis blog for the 2022 State of Rust Survey. Please discuss any thoughts, questions, comments here. This blog was drafted by the Rust Foundation with input and oversight by @rylev & @nrc. 

Please note that we initially prepared an additional blog to serve as the vehicle to download the report. Our plan had been for that additional blog to be published first with this one to follow. 

Given the timing, it might be best to consolidate both blogs into one by simply adding a download call-to-action/link to this post, we welcome your thoughts.

### CODE REVIEW

File: posts/2023-state-of-rust-analysis.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Before diving into a few highlights, we would like to thank everyone who was involved in creating the State of Rust survey with special acknowledgment to the translators whose work allowed us to offer the survey in English, Simplified Chinese, Traditional Chinese, French, German, Japanese, Korean, Portuguese, Russian, Spanish, and Ukrainian. 
+
+## Participation
+In 2022, we had 9,433 total survey completions and an increased survey completion rate of 82% vs. 76% in 2021. While the goal is always total survey completion for all participants, the survey requires time, energy, and focus – we consider this figure quite high and were pleased by the increase. 
+
+
+We also saw a significant increase in the number of people viewing but not participating in the survey (from 16,457 views in 2021 to 2,5581 – a view increase of over 55%). While this is likely due to a number of different factors, we feel this information speaks to the rising interest in Rust and the growing general audience following its evolution.
```
Comment (ehuss) [inline posts/2023-state-of-rust-analysis.md:20]: I assume:

```suggestion
We also saw a significant increase in the number of people viewing but not participating in the survey (from 16,457 views in 2021 to 25,581 – a view increase of over 55%). While this is likely due to a number of different factors, we feel this information speaks to the rising interest in Rust and the growing general audience following its evolution. 
```

**Review by ehuss — COMMENTED** [review]
Thanks for getting this posted!

Just FYI, the filename needs to include the month and day of publication for the site generator to work.

Also, the images are too blurry for me to read them.

File: posts/2023-state-of-rust-analysis.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+![5](https://github.com/rust-lang/blog.rust-lang.org/assets/45044840/c533954b-eadc-41c0-ac9e-717d7ebc6e22)
+
+It is exciting to see the continued growth of professional Rust usage and the confidence so many users feel in its performance, control, security and safety, enjoyability, and more!
+
+## Supporting the Future of Rust
+
+A key goal of the State of Rust survey is to shed light on challenges, concerns, and priorities Rustaceans are currently sitting with. 
+
+Of those respondents who shared their main worries for the future of Rust, 26% have concerns that the developers and maintainers behind Rust are not properly supported – a decrease of more than 30% from the previous year’s findings. While we would of course like to see this number at 0%, this improvement reflects the growing maturity of the Rust Foundation and the support they provided the Project in 2022 during its first full year in operation.
```
Comment (compiler-errors) [inline posts/2023-state-of-rust-analysis.md:65]: I'm not sure if this is the right channel to note this, but this statement:

> 26% have concerns that the developers and maintainers behind Rust are not properly supported – a decrease of more than 30% from the previous year’s findings.

Does not necessarily lead to this conclusion:

> this improvement reflects the growing maturity of the Rust Foundation and the support they provided the Project in 2022 during its first full year in operation.

While certainly the Foundation's involvement in supporting Rust maintainers and project members is important and shouldn't be understated, I was surprised by the wording the first time I read through this. There are other reasons why this number may have dropped -- for example, this is gauging sentiment from the Rust community at large, and not (if I'm understanding the data correctly) reporting the feelings of Rust project members and Rust maintainers themselves (who are a small percentage of this population as a whole). Also, project members and maintainers do receive other channels of support, such as partial or full compensation to work on Rust-related activities from employers and non-employer sponsorship such as Github sponsors and patreon.

Given all of that, I think this should be weakened to not make such a direct claim. Perhaps it's better to just use wording like "suggests", or back it up with other data that leads to this specific conclusion.

File: posts/2023-state-of-rust-analysis.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title: 2022 Annual Rust Survey Results
+author: The Rust Survey Working Group in partnership with the Rust Foundation
+release: false
+---
+
+Hello, Rustaceans!
+
+For the 6th year in a row, the Rust Project conducted a survey on the Rust programming language, with participation from project maintainers, contributors, and those generally interested in the future of Rust. This edition of the annual State of Rust Survey opened for submissions on December 5 and ran until December 22, 2022. Thank you for your patience as we’ve worked to assemble this analysis of the survey responses. We plan to identify a more expedient and sustainable process going forward.
```
Comment (nikomatsakis) [inline posts/2023-state-of-rust-analysis.md:10]: Is there a more detailed analysis of the data available? Specifically I was wondering whether there is a simple list of the questions + the aggregate results for each question on the github repo. If so, it'd be good to link to it.

File: posts/2023-state-of-rust-analysis.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+The vast majority of our respondents reported being most comfortable communicating on technical topics in English (93%), followed by Chinese (7%). 
+
+Rust user respondents were asked which country they live in. The top 13 countries represented were as follows: United States (25%), Germany (12%), China (7%), United Kingdom (6%), France (5%), Canada (4%), Russia (4%), Japan (3%), Netherlands (3%), Sweden (2%), Australia (2%), Poland (2%), India (2%). Nearly 72.5% of respondents elected to answer this question. 
+
+While we see global access to Rust education as a critical goal for our community, we are proud to say that Rust was used all over the world in 2022!
+
+
+## Rust Usage
+More people are using Rust than ever before! Of our respondents using Rust, 47% do so on a daily basis – an increase of 4% from the previous year.
```
[the remark this answers is not in the document]
Comment (nikomatsakis) [reply posts/2023-state-of-rust-analysis.md:35]: Seems like yes -- I would have liked to see that overall number cited here.

**Review by nikomatsakis — COMMENTED** [review]
I'm excited to see this post, @graciegregory! I enjoyed reading it. 

I left a few small suggestions on the phrasing or additional information I thought would be helpful.

File: posts/2023-state-of-rust-analysis.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+The goal of this survey is always to give our wider community a chance to express their opinions about the language we all love and help shape its future. We’re grateful to those of you who took the time to share your voice on the state of Rust last year. 
+
+Before diving into a few highlights, we would like to thank everyone who was involved in creating the State of Rust survey with special acknowledgment to the translators whose work allowed us to offer the survey in English, Simplified Chinese, Traditional Chinese, French, German, Japanese, Korean, Portuguese, Russian, Spanish, and Ukrainian. 
+
+## Participation
+In 2022, we had 9,433 total survey completions and an increased survey completion rate of 82% vs. 76% in 2021. While the goal is always total survey completion for all participants, the survey requires time, energy, and focus – we consider this figure quite high and were pleased by the increase. 
+
+
```
Comment (rylev) [inline posts/2023-state-of-rust-analysis.md:19]:
```suggestion
```

File: posts/2023-state-of-rust-analysis.md (hunk 3 above)
Comment (rylev) [inline posts/2023-state-of-rust-analysis.md:10]: Perhaps we should add a link to the announcement blog post.

```suggestion
For the 6th year in a row, the Rust Project [conducted a survey on the Rust programming language](https://blog.rust-lang.org/2022/12/05/survey-launch.html), with participation from project maintainers, contributors, and those generally interested in the future of Rust. This edition of the annual State of Rust Survey opened for submissions on December 5 and ran until December 22, 2022. Thank you for your patience as we’ve worked to assemble this analysis of the survey responses. We plan to identify a more expedient and sustainable process going forward. 
```

Comment (rylev) [reply posts/2023-state-of-rust-analysis.md:10]: I don't think we have that now. While it's certainly possible, I think it might take a bit of work to get into a format that is legible for general consumption. I wouldn't block this (long overdue) post on that, but we should try our best to get something like that out ASAP (cc @nrc).

**Comment by kettle11** [comment]
Previous Rust survey summaries went into far more detail on Rust user opinion of technical aspects like compile times, library support, compiler errors, IDE integration etc. 
 
If you look at the "Challenges Ahead" section from last year's survey there's far more precise detail and about what areas improved / still need improvement: https://blog.rust-lang.org/2022/02/15/Rust-Survey-2021.html

Looking at this draft it looks like there's very little of that in this year's survey summary. I'd really like to see that improved as that info helps the community know how Rust is doing / what still needs to be improved.

File: posts/2023-state-of-rust-analysis.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+## Rust Usage
+More people are using Rust than ever before! Of our respondents using Rust, 47% do so on a daily basis – an increase of 4% from the previous year. 
+
+30% of Rust user respondents can write simple programs in Rust, 27% can write production-ready code, and 42% consider themselves productive using Rust. 
+
+Over 90% of survey respondents identified as Rust users. Of the former Rust users who completed the survey, 30% cited difficulty as the primary reason for giving up while nearly 47% cited factors outside of their control. 
+
+![1](https://github.com/rust-lang/blog.rust-lang.org/assets/45044840/54c56ffc-2a76-4144-9800-94f5e7da0180)
```
Comment (Turbo87) [inline posts/2023-state-of-rust-analysis.md:40]: I think something might have gone wrong with these images since they are somehow not part of the pull request diff. I assume they should use relative paths to https://github.com/rust-lang/blog.rust-lang.org/tree/master/static/images?

@graciegregory let me know if you need help with that :)

**Comment by rylev** [comment]
I agree that the analysis from this year lacks a bit of detail on what exact challenges lie ahead for the Rust Project and community. I think that's due to the fact that it's a bit more ambiguous what those challenges actually are. That coupled with the fact that Rust Project leadership has been in a state of transition means that they results aren't particularly clear and its hard to draw clear insights from them. I think the higher order bit is that the community didn't give any strong indication of one particular clear area of concern. 

I think we should merge this as soon as the issue with the images is fixed. While it's not the most satisfying survey results post we've had, it's very delayed, and I think it'd be better to just get it out there.

**Comment by kettle11** [comment]
> I think that's due to the fact that it's a bit more ambiguous what those challenges actually are.

Is that not the purpose of the survey? Previous years touched on community opinion of compile times, disk space, debugging tools, IDE integration, GUI development, etc. All important topics!

Those questions were still asked this year, no? It feels important to have continuity in that discussion.

I'm pushing back because often during times of new leadership / transition organizations like to switch into "good news only" mode, but an open-source project like Rust should maintain transparency. A frank appraisal of technical issues, whether good or bad, is important for the community to understand how Rust is doing.

**Review by rylev — APPROVED** [review]
